### PR TITLE
Delete associated indexes when deleting coordinate variables.

### LINKED ISSUE #3746: dataarray arithmetics restore removed coordinates in xarray 0.15
#### MCVE Code Sample

```python
import xarray as xr
import numpy as np
arr2 = xr.DataArray(np.ones((2, 2)), dims=['y', 'x'])
arr1 = xr.DataArray(np.ones((2, 2)), dims=['y', 'x'], coords={'y': [0, 1], 'x': [0, 1]})

del arr1.coords['y']
del arr1.coords['x']

# shows arr1 without coordinates
arr1

# shows coordinates in xarray 0.15
arr1 * arr2

```

#### Expected Output
```python
<xarray.DataArray (y: 2, x: 2)>
array([[1., 1.],
       [1., 1.]])
Dimensions without coordinates: y, x
```

#### Problem Description
In xarray 0.15, the coordinates are restored when doing the multiplication:
```python
<xarray.DataArray (y: 2, x: 2)>
array([[1., 1.],
       [1., 1.]])
Coordinates:
  * y        (y) int64 0 1
  * x        (x) int64 0 1
```

#### Output of ``xr.show_versions()``
<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.8.1 | packaged by conda-forge | (default, Jan 29 2020, 14:55:04) 
[GCC 7.3.0]
python-bits: 64
OS: Linux
OS-release: 4.18.0-147.0.3.el8_1.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8
LOCALE: en_GB.UTF-8
libhdf5: 1.10.5
libnetcdf: 4.7.3

xarray: 0.15.0
pandas: 1.0.0
numpy: 1.18.1
scipy: 1.4.1
netCDF4: 1.5.3
pydap: None
h5netcdf: 0.7.4
h5py: 2.10.0
Nio: None
zarr: 2.3.2
cftime: 1.0.4.2
nc_time_axis: None
PseudoNetCDF: None
rasterio: 1.1.2
cfgrib: None
iris: None
bottleneck: 1.3.1
dask: 2.10.1
distributed: 2.10.0
matplotlib: 3.1.3
cartopy: 0.17.0
seaborn: None
numbagg: None
setuptools: 45.1.0.post20200119
pip: 20.0.2
conda: None
pytest: 5.3.5
IPython: 7.12.0
sphinx: 2.3.1

</details>

### PR DESCRIPTION
**Pull request by dcherian**
- [x] Closes #3746
 - [x] Tests added
 - [x] Passes `isort -rc . && black . && mypy . && flake8`
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

@shoyer do you think this is the right thing to do?

### CODE REVIEW

**Comment by mathause** [comment]
Potentially a fix for #3739

File: xarray/core/coordinates.py [hunk 1]
```diff
@@ -246,8 +246,12 @@ def _update_coords(
     def __delitem__(self, key: Hashable) -> None:
         if key in self:
             del self._data[key]
+            if key in self._data.indexes:
+                indexes = dict(self._data.indexes)
+                indexes.pop(key)
+                self._data._indexes = indexes
```
Comment (shoyer) [inline xarray/core/coordinates.py]: `Dataset.__delitem__` already removes indexes, so I believe this is redundant.

File: xarray/core/coordinates.py [hunk 2]
```diff
@@ -312,7 +316,14 @@ def to_dataset(self) -> "Dataset":
         return Dataset._construct_direct(coords, set(coords))
 
     def __delitem__(self, key: Hashable) -> None:
-        del self._data._coords[key]
+        if key in self:
+            del self._data._coords[key]
+            if key in self._data.indexes:
+                indexes = dict(self._data.indexes)
+                indexes.pop(key)
+                self._data._indexes = indexes
```
Comment (shoyer) [inline xarray/core/coordinates.py]: `_indexes` should be unique to each `DataArray` object, so it's fine to update this in-place like what we do on `Dataset.__delitem__`:
```suggestion
                del self._data._indexes[key]
```

File: xarray/core/coordinates.py (hunk 1 above)
Comment (dcherian) [reply xarray/core/coordinates.py]: :+1:

File: xarray/core/coordinates.py (hunk 2 above)
Comment (dcherian) [reply xarray/core/coordinates.py]: :+1:

**Comment by max-sixty** [comment]
OK to merge @dcherian ? We can almost release 0.15.1 then

**Comment by dcherian** [comment]
I think so...

**Comment by max-sixty** [comment]
I'm hitting the big green button! 😅

**Pull request merged**